Automattic/wp-calypso
# Reader: Reimplement keyboard shortcuts

#### Changes proposed in this Pull Request

This PR aims to reintroduce a simpler version of the keyboard shortcuts that were supported in Reader and were removed in #60501. See https://github.com/Automattic/wp-calypso/pull/60501#issuecomment-1028083347 and https://github.com/Automattic/wp-calypso/pull/60501#issuecomment-1028593376 for the motivation.

The PR is essentially reverting parts of #60501 that were related to the Reader, but also update the keyboard shortcut handling to happen directly in the components, rather than using the legacy removed keyboard shortcuts module.

~The PR is still in progress and contains a commit from #60800, so we have to rebase it after it lands. Also, I'd like to test all scenarios, so this PR isn't ready for review just yet.~ I've decided to untangle this from #60800 and land it separately due to popular demand 😛 

cc @sirbrillig @pento for awareness, and also if you'd like to give it a spin earlier.

#### Testing instructions

* Keep an eye on the console for errors while testing this.
* Go to `/read`
* Test the following keyboard shortcuts and verify they work well:
  * `j` - move selection down
  * `k` - move selection up
  * `Enter/Return` - open the full post view for the selected post
  * `v` - open the selected post in a new tab
  * `l` (lowercase L) - toggle like/unlike the selected post
  * `.` - go to top - works only if you have new updates in the stream.
* Open a selected post in a full post view
* Test the following keyboard shortcuts and verify they work well:
  * `Esc` - close full post view
  * `l` (lowercase L) - toggle like/unlike the post
  * `j` - previous post
  * `k` - next post
